File: Example/ios/iOS Example/OptOutViewController.m
//
//  OptOutViewController.m
//  PiwikTracker
//
//  Created by Mattias Levin on 9/17/13.
//  Copyright (c) 2013 Mattias Levin. All rights reserved.
//

#import "OptOutViewController.h"
#import "PiwikTracker.h"


@interface OptOutViewController ()
@end


@implementation OptOutViewController


- (void)viewDidLoad {
  [super viewDidLoad];
  
  self.optOutSwitch.on = ![PiwikTracker sharedInstance].optOut;
  
}


- (IBAction)sendOptOutAction:(id)sender {
  
  UISwitch *option = (UISwitch*)sender;
  
  if (option.on) {
    
    [PiwikTracker sharedInstance].optOut = NO;
    
  } else {
    
    [PiwikTracker sharedInstance].optOut = YES;
    
  }
  
}


@end

File: Example/ios/iOS Example/OptOutViewController.swift
import UIKit
import MatomoTracker

class OptOutViewController: UIViewController {
    
    @IBOutlet weak var optOutSwitch: UISwitch!
    @IBAction func optOutSwitchChanged(_ sender: Any) {
        MatomoTracker.shared.isOptedOut = optOutSwitch.isOn
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        optOutSwitch.isOn = MatomoTracker.shared.isOptedOut
    }
    
    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
        MatomoTracker.shared.track(view: ["menu","opt out"])
    }
}
